Python/bytecodes.c

```diff
@@ -0,0 +1,3983 @@
+            assert(oparg != 0);
+            PyObject *peek = PEEK(oparg);
+            Py_INCREF(peek);
+            PUSH(peek);
+        }
+
+        // stack effect: (__0 -- )
+        inst(BINARY_OP) {
+            PREDICTED(BINARY_OP);
+            PyObject *rhs = POP();
+            PyObject *lhs = TOP();
+            assert(0 <= oparg);
+            assert((unsigned)oparg < Py_ARRAY_LENGTH(binary_ops));
+            assert(binary_ops[oparg]);
+            PyObject *res = binary_ops[oparg](lhs, rhs);
+            Py_DECREF(lhs);
+            Py_DECREF(rhs);
+            SET_TOP(res);
+            if (res == NULL) {
+                goto error;
+            }
+            JUMPBY(INLINE_CACHE_ENTRIES_BINARY_OP);
+        }
+
+        // stack effect: (__0 -- )
+        inst(BINARY_OP_ADAPTIVE) {
+            assert(cframe.use_tracing == 0);
+            _PyBinaryOpCache *cache = (_PyBinaryOpCache *)next_instr;
+            if (ADAPTIVE_COUNTER_IS_ZERO(cache)) {
+                PyObject *lhs = SECOND();
+                PyObject *rhs = TOP();
+                next_instr--;
+                _Py_Specialize_BinaryOp(lhs, rhs, next_instr, oparg, &GETLOCAL(0));
+                DISPATCH_SAME_OPARG();
+            }
+            else {
+                STAT_INC(BINARY_OP, deferred);
+                DECREMENT_ADAPTIVE_COUNTER(cache);
+                GO_TO_INSTRUCTION(BINARY_OP);
+            }
+        }
+
+        // stack effect: ( -- )
+        inst(SWAP) {
+            assert(oparg != 0);
+            PyObject *top = TOP();
+            SET_TOP(PEEK(oparg));
+            PEEK(oparg) = top;
+        }
+
+        // stack effect: ( -- )
+        inst(EXTENDED_ARG) {
+            assert(oparg);
+            oparg <<= 8;
+            oparg |= _Py_OPARG(*next_instr);
+            // We might be tracing. To avoid breaking tracing guarantees in
+            // quickened instructions, always deoptimize the next opcode:
+            opcode = _PyOpcode_Deopt[_Py_OPCODE(*next_instr)];
+            PRE_DISPATCH_GOTO();
+            // CPython hasn't traced the following instruction historically
+            // (DO_TRACING would clobber our extended oparg anyways), so just
+            // skip our usual cframe.use_tracing check before dispatch. Also,
+            // make sure the next instruction isn't a RESUME, since that needs
+            // to trace properly (and shouldn't have an extended arg anyways):
+            assert(opcode != RESUME);
+            DISPATCH_GOTO();
+        }
+
+        // stack effect: ( -- )
+        inst(EXTENDED_ARG_QUICK) {
+            assert(cframe.use_tracing == 0);
+            assert(oparg);
+            int oldoparg = oparg;
+            NEXTOPARG();
+            oparg |= oldoparg << 8;
+            DISPATCH_GOTO();
+        }
+
+        // stack effect: ( -- )
+        inst(CACHE) {
+            Py_UNREACHABLE();
+        }
+
+
+// END BYTECODES //
+
+    }
+ handle_eval_breaker:;
+ unbound_local_error:;
+ error:;
```

Comment:
These semicolons aren't needed, right?

```suggestion
 handle_eval_breaker:
 unbound_local_error:
 error:
```

Reply:
Oh, you're right. It'll have to wait until another time. (The _last_ label does need the `;` though, a label must be followed by a statement.)